scripts/build.sh

```diff
@@ -26,25 +26,25 @@ rm -rf pkg/*
 mkdir -p bin/
 
 # If its dev mode, only build for ourself
-if [ "${TF_DEV}x" != "x" ]; then
+if [[ -n "${TF_DEV}" ]]; then
     XC_OS=$(go env GOOS)
     XC_ARCH=$(go env GOARCH)
+
+    # Allow LD_FLAGS to be appended during development compilations
+    LD_FLAGS="-X main.GitCommit=${GIT_COMMIT}${GIT_DIRTY} $LD_FLAGS"
 fi
 
 if ! which gox > /dev/null; then
     echo "==> Installing gox..."
     go get -u github.com/mitchellh/gox
 fi
 
-# instruct gox to build statically linked binaries
+# Instruct gox to build statically linked binaries
 export CGO_ENABLED=0
 
-# Allow LD_FLAGS to be appended during development compilations
-LD_FLAGS="-X main.GitCommit=${GIT_COMMIT}${GIT_DIRTY} $LD_FLAGS"
-
 # In release mode we don't want debug information in the binary
 if [[ -n "${TF_RELEASE}" ]]; then
-    LD_FLAGS="-X main.GitCommit=${GIT_COMMIT}${GIT_DIRTY} -X github.com/hashicorp/terraform/version.Prerelease= -s -w"
+    LD_FLAGS="-s -w"
```

Comment:
Could we not just remove the `github.com/hashicorp/terraform/version.Prerelease` bit here and leave everything else unchanged? I don't see how setting `main.GitCommit` is related to setting `version.Prerelease` :thinking: 

I don't think there's really any strong reason for us to be overriding `version.Prerelease` here anyway, since we manually update the source file `version.go` to have an empty string for this as part of the stable release process anyway.

Probably this ended up here as part of introducing the release automation that we now use for providers, but we don't use that for core so I think we don't need this here right now and we'll be revisiting the automated release process for Terraform core in the future anyway, at which point we can figure out what we want to do here for the long term.

Reply:
Without the additional changes `terraform version` would show `Terraform v0.11.12-beta1 (xxxxxxxxxxxxxxxxxxxxxxxxx)` which I thought didn't make sense for a proper prerelease (it would do that for all prereleases, so also RCs). So I limited setting the commit SHA for dev builds only.

The other solution to prevent showing the commit SHA for prereleases would require changes to the `version` command. That's also possible of course, but I just didn't see the added value for having the commit SHA set for non dev builds.

Reply:
Including the git commit in actual releases doesn't seem necessarily _bad_ to me, but indeed it is a change of behavior so better to save that for another time if we want to make that change.

(I've been considering also including some other info in `terraform version`, like `GOOS` and `GOARCH`, so we get more context when people open bug reports, but that's all for another day.)

Reply:
That would indeed be very nice, to output more context that we could request by default when opening an issue or bug report.

But maybe that should be behind an optional flag (e.g `terraform version -v`) to not make the default version command too chatty? But indeed that feels like a separate change.